gamemode/core/hooks/sv_hooks.lua

```diff
@@ -118,10 +118,10 @@ function GM:CanPlayerInteractItem(client, action, item)
 end
 
 function GM:CanPlayerTakeItem(client, item)
-	if (type(item) == "Entity") then
+	if (type(item.entity) == "Entity") then
```

Comment:
Maybe change this to `IsValid(item.entity)` in case `item.entity` is set to `NULL`.